File: java/com/dragon/inter/Or.java
package com.dragon.inter;

import com.dragon.lexer.Token;

public class Or extends Logical {
  public Or(Token tok, Expr left, Expr right) {
    super(tok, left, right);
  }

  @Override
  public void jumping(int to, int from) {
    var label = to != 0 ? to : newLabel();
    left.jumping(label, 0);
    right.jumping(to, from);
    if (to == 0) {
      emitLabel(label);
    }
  }
}

File: cpp/inter/or.cpp
#include "cpp/inter/or.hpp"

#include "cpp/inter/node.hpp"

namespace inter {
void Or::jumping(std::stringstream &ss, std::uint32_t to, std::uint32_t from) {
  int label = to != 0 ? to : new_label();
  expr1()->jumping(ss, label, 0);
  expr2()->jumping(ss, to, from);
  if (!to) {
    emit_label(ss, label);
  }
}
}  // namespace inter
